foyer-storage/src/device/direct_file.rs

```diff
@@ -123,14 +130,22 @@ impl DirectFileDevice {
             buf.set_len(aligned);
         }
 
-        let file = self.file.clone();
+        let mut file = self.file.clone();
         let mut buffer = asyncify_with_runtime(&self.runtime, move || {
-            #[cfg(target_family = "unix")]
-            use std::os::unix::fs::FileExt;
             #[cfg(target_family = "windows")]
-            use std::os::windows::fs::FileExt;
+            let read = {
+                let original = file.stream_position()?;
+                file.seek(std::io::SeekFrom::Start(offset))?;
+                let read = file.read(buf.as_mut())?;
+                file.seek(std::io::SeekFrom::Start(original))?;
```

Comment:
We can skip this if you are sure that we don't need to seek the file to beginning because this is trying to emulate read_at behavior.

Reply:
Hi, @spector-9 . I'm not familar with Windows API. On Linux, positioned read/write will bypass  modification of fd, which is lighter and faster.

foyer aims to be a hybrid cache with optimal performance. seek io is not acceptable. (BTW, I think there is concurrent bug without a lock to protect seek and io)

Reply:
@MrCroxx Hi, 
That's why I tried to keep unix and windows separate. On windows there's no read_at so there's no other option than to seek and call synchronous_read under the hood also synchronous_read moves the cursor to whatever happens to be the last position. 
To be more ergonomic I can call seek_read rather than manually seeking and then reading. I might also be able to remove the final seek that move the cursor to original position if it's not needed.

Reply:
Oh, sorry about that. Seems my ios safari browser plugin distribes the github diff and I mistook the add and remove (realky wired...). I'll do a careful review tomorrow.

Reply:
It revented the color in dark mode. 🤣